pypdf/filters.py

```diff
@@ -152,6 +147,7 @@ def decode(
                 colors = cast(int, decode_parms[LZW.COLORS].get_object())  # type: ignore
             except (TypeError, KeyError):
                 colors = 1
+            DEFAULT_BITS_PER_COMPONENT = 8
```

Comment:
Defining this inside the method seems odd at all. I would propose making this an internal class-level constant instead.

Reply:
Magic constants tend to be discouraged, thus I am not sure whether your proposed solution fits here.